9-ое ТЗ.

nev-ivan opened the pull request.

Commit:
Первая итерация 9-го ТЗ.

Commit:
Первая итерация 9-го ТЗ.

Ivanov-s-tmn commented on java-kanban.iml:
.iml файл стоит добавить в исключения gitignore, в нем содержится локальная конфигурация, которая не нужна в репозитории

Ivanov-s-tmn commented on src/handler/EpicsHandler.java:
```diff
@@ -0,0 +1,83 @@
+import java.io.IOException;
+import java.util.List;
+
+public class EpicsHandler extends BaseHttpHandler implements HttpHandler {
```
Можно имплементировать интерфейс `HttpHandler` в `BaseHttpHandler` и сделать его абстрактным, плюс, вынести в него инициализацию таск менеджера и gson, таким образом не будет дублирования в хендлерах-реализациях

Ivanov-s-tmn commented on src/manager/TaskValidationException.java:
```diff
@@ -0,0 +1,4 @@
+package manager;
+
+public class TaskValidationException extends Throwable {
```
Свои исключения стоит делать непроверяемыми `RuntimeException` тогда компилятор не обяжет нас их обработать и мы можем их использовать по нашей необходимости в логике

Ivanov-s-tmn commented on test/http/HttpTaskManagerTest.java:
Тестами нужно покрыть все эндпоинты и лучше разделить тесты по хендлерам

Commit:
Вторая итерация 9-го ТЗ.
- Тесты пакета http разделить по хэндлерам
- Собственное исключение изменить на непроверяемое
- Добавить .iml в gitignore
- Класс BaseHttpHandler сделать абстрактным и имплементировать от него

Ivanov-s-tmn commented on src/handler/EpicsHandler.java:
```diff
@@ -26,17 +20,17 @@ public void handle(HttpExchange exchange) throws IOException {
                     final List<SubTask> subtasks = epic.getSubTasksIds().stream()
-                            .map(taskManager::getSubTask)
+                            .map(manager::getSubTask)
                             .toList();
```
В таск менеджере  же есть метод для получения сабтасок по айди эпика.  Такая реализация не подходит, так как `manager::getSubTask` будет добавлять сабтаски в историю, а в этом случае это некорректно

Ivanov-s-tmn commented on src/handler/EpicsHandler.java:
```diff
@@ -26,17 +20,17 @@ public void handle(HttpExchange exchange) throws IOException {
         switch (exchange.getRequestMethod()) {
             case "GET":
-                final EpicTask epic = taskManager.getEpicTask(idFromRequest);
+                final EpicTask epic = manager.getEpicTask(idFromRequest);
```
Этот метод нужно дернуть только в случае, если запрос придет на эндпоинт `/subtasks/{id}`, иначе будет лишний вызов и добавление в историю

Commit:
Третья итерация 9-го ТЗ.
- Исправить работу EpicsHandler
